src/SmartComponents/BaselinesPage/BaselinesKebab/BaselinesKebab.js

```diff
@@ -1,78 +1,86 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { withRouter } from 'react-router-dom';
 
-import { baselinesKebabActions } from './redux';
 import { Dropdown, KebabToggle, DropdownItem } from '@patternfly/react-core';
+import DeleteBaselinesModal from '../DeleteBaselinesModal/DeleteBaselinesModal';
 
 class BaselinesKebab extends Component {
     constructor(props) {
         super(props);
 
         this.state = {
-            kebabOpened: false
+            kebabOpened: false,
+            modalOpened: false
         };
 
-        this.exportSelect = this.exportSelect.bind(this);
-
         this.toggleKebab = () => {
             const { kebabOpened } = this.state;
             this.setState({
                 kebabOpened: !kebabOpened
             });
         };
-    }
 
-    exportSelect() {
-        const { exportType, baselineTableData, baselineData, baselineRowData } = this.props;
-        this.toggleKebab();
-        if (exportType === 'baseline list') {
-            this.props.exportToCSV(exportType, baselineTableData);
-        } else if (exportType === 'baselines data') {
-            this.props.exportToCSV(exportType, baselineData, baselineRowData);
-        }
+        this.toggleModalOpened = () => {
+            const { modalOpened } = this.state;
+            this.setState({
+                modalOpened: !modalOpened
+            });
+        };
     }
 
     render() {
-        const { kebabOpened } = this.state;
-        const dropdownItems = [
-            <DropdownItem key="export" component="button" onClick={ this.exportSelect }>Export as CSV</DropdownItem>
-        ];
+        const { kebabOpened, modalOpened } = this.state;
+        const { selectedBaselineIds } = this.props;
+        let dropdownItems;
+
+        if (selectedBaselineIds.length > 0) {
+            dropdownItems = [
+                <DropdownItem
+                    key="multi-delete"
+                    component="button"
+                    onClick={ this.toggleModalOpened }
+                >
+                    Delete baselines
+                </DropdownItem>
+            ];
+        } else {
+            dropdownItems = [
+                <DropdownItem
+                    key="multi-delete"
+                    component="button"
+                    onClick={ this.toggleModalOpened }
+                    isDisabled
+                >
+                    Delete baselines
+                </DropdownItem>
+            ];
```

Comment:
You can simplify this by dropping the if/else and doing:

```
           dropdownItems = [
                <DropdownItem
                    key="multi-delete"
                    component="button"
                    onClick={ this.toggleModalOpened }
                    isDisabled={ selectedBaselineIds.length > 0 }
                >
                    Delete baselines
                </DropdownItem>
            ];
```

Note how the `isDisabled` takes a value and can be set based on a condition reducing the repetition.

Reply:
I didn't know that! I can fix a number of things with this. It makes sense how that would work though.